api/app/Models/Activity.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace App\Models;
+
+use Database\Helpers\TeamHelpers;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Support\Facades\Auth;
+use Spatie\Activitylog\Models\Activity as SpatieActivity;
+
+class Activity extends SpatieActivity
+{
+    protected $keyType = 'string';
+
+    protected function properties(): Attribute
+    {
+        return Attribute::get(function ($value) {
+            $data = is_array($value) ? $value : json_decode($value, true);
+
+            return [
+                'attributes' => $data['attributes'] ?? null,
+                'old' => $data['old'] ?? null,
+            ];
+        });
+    }
+
+    public function scopeWhereIsPoolActivity(Builder $query)
+    {
+        return $query->where('subject_type', Pool::class);
+    }
```

Comment:
Do we want to append some more statements like causer is non-null and a user? Or is that just pointless/overcaution?